util/rom_chip_info.py

```diff
@@ -43,11 +45,14 @@ def generate_chip_info_c_source(scm_revision: int) -> str:
 """
 
 
-def read_version_file(path: Path) -> int:
-    """Interprets the contents of `path` as a big-endian hex literal."""
+def read_version_file(version_info) -> int:
+    """
+    Search for the BUILD_SCM_REVISION variable and interprets
+    the contents as a big-endian hex literal. Return a constant
+    if the variable is not found.
+    """
 
-    with open(path, "rt") as f:
-        version = f.read().strip()
+    version = version_info.get('BUILD_SCM_REVISION', "8badF00d")
```

Comment:
I think I'd prefer the current approach where a version file is produced to them, instead of having all these tools having to parse volatile-status.txt.

I think having an extra step in the middle (like we currently do), have an advantage that if we need stamping, we can still have some cache hit if some bits of volatile-status.txt changed, but the needed bits happen to stay the same.

Reply:
At the moment though, we have a mix of tools using those "derived" files and some other parsing volatile-status.txt (or a copy thereof). This also means that we would have to use "fake" values when stamping is disabled which I don't find to be particularly useful.
Also, some tools like reggen use at least four different pieces of information so that would require to pass four different files on the command-line which I am not sure is much better than passing a single file with a known format?
What I am proposing is that all tools use the same format which is parsed by the same code contained in a python library so there is no code duplication.